crates/rome_js_analyze/src/analyzers/nursery/no_sequences.rs

```diff
@@ -0,0 +1,84 @@
+use rome_analyze::context::RuleContext;
+use rome_analyze::{declare_rule, Ast, Rule, RuleDiagnostic};
+use rome_js_syntax::{JsForStatement, JsSequenceExpression};
+use rome_rowan::AstNode;
+
+declare_rule! {
+    /// Disallow comma operator.
+    ///
+    /// The comma operator includes multiple expressions where only one is expected.
+    /// It evaluates every operand from left to right and returns the value of the last operand.
+    /// It frequently obscures side effects, and its use is often an accident.
+    ///
+    /// The use of the coma operator in the initialization and update parts of a `for` is still allowed.
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-sequences
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = doSomething(), 0;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// for (; doSomething(), !!test; ) {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// // Use a semicolon instead.
+    /// let a, b;
+    /// a = 1, b = 2;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// for(a = 0, b = 0; (a + b) < 10; a++, b += 2) {}
+    /// ```
+    ///
+    pub(crate) NoSequences {
+        version: "12.0.0",
+        name: "noSequences",
+        recommended: true,
+    }
+}
+
+impl Rule for NoSequences {
+    type Query = Ast<JsSequenceExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let seq = ctx.query();
+        if let Some(for_stmt) = seq.parent::<JsForStatement>() {
+            if for_stmt
+                .initializer()
+                .filter(|for_init| for_init.syntax() == seq.syntax())
+                .is_some()
+                || for_stmt
+                    .update()
+                    .filter(|for_up| for_up.syntax() == seq.syntax())
+                    .is_some()
```

Comment:
I think this should work too
```suggestion
            if for_stmt.initializer() == Some(
                .filter(|for_init| for_init.syntax() == seq.syntax())
                .is_some()
                || for_stmt
                    .update()
                    .filter(|for_up| for_up.syntax() == seq.syntax())
                    .is_some()
```

Or you can test if the expression isn't the `test` expression (it then is guaranteed to be the update or init):

```
for_stmt.test().map(AstNode::into_syntax).as_ref() == Some(seq.syntax())
```